semantiva/pipeline/nodes/nodes.py

```diff
@@ -644,6 +645,74 @@ def get_created_keys(cls) -> List[str]:
         return cls.processor.get_created_keys()
 
 
+class DataOperationContextInjectorProbeNode(DataOperationNode):
+    """A node that runs a :class:`DataOperation`, stores its output in the
+    context under a specified keyword, and forwards the original data."""
+
+    context_keyword: str
+
+    def __init__(
+        self,
+        processor: Type[BaseDataProcessor],
```

Comment:
I would restrict to `DataOperation`. The Base also implies putting a ProbeNode
```suggestion
        processor: Type[DataOperation],
```